src/main/kotlin/com/coder/gateway/CoderRemoteConnectionHandle.kt

```diff
@@ -200,5 +238,66 @@ class CoderRemoteConnectionHandle {
             }
             return Pair(tokenFromUser, tokenSource)
         }
+
+        /**
+         * Return if the URL is whitelisted, https, and the URL and its final
+         * destination, if it is a different host.
+         */
+        @JvmStatic
+        fun isWhitelisted(url: URL, deploymentURL: URL): Triple<Boolean, Boolean, String> {
+            // TODO: Setting for the whitelist, and remember previously allowed
+            //  domains.
+            val domainWhitelist = listOf("intellij.net", "jetbrains.com", deploymentURL.host)
```

Comment:
`deploymentURL.host`

Ah interesting - do we provide a way to serve Intellij IDEs from Coder?
I could imagine the deployment host being `coder.acme.com` and downloading IDEs from `tools.acme.com`.  Of course, this becomes more of a moot point if the domain allowlist can be configured.

Reply:
I think it was premature of me to add the deployment host.  I know some folks self-serve IDEs bu they are probably going to either use a different domain or the same root domain like in your example, but definitely not the exact same sub-domain as Coder.

So I will remove the deployment host for now I think.

Reply:
Actually might add localhost for development and because I have seen some folks do it that way to make the deploy completely offline.